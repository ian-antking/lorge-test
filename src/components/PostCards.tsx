import PostCard from './PostCard'
import styled from 'styled-components'

interface Post {
  id: string
  title: string
  author: string
  blurb: string
  published: string
}

interface Props {
  posts: Post[]
}

const PostCardContainer = styled.div`
  width: 100vw;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
`

const PostCards: React.FC<Props> = ({ posts }): React.ReactElement => {
  if (posts.length > 0) {
    return (
      <PostCardContainer>
        {posts.map((post: Post) => (
          <PostCard
            key={post.id}
            id={post.id}
            title={post.title}
            author={post.author}
            published={post.published}
            blurb={post.blurb}
          />
        ))}
      </PostCardContainer>
    )
  } else {
    return (
      <PostCardContainer>
        <h1>Nothing Here</h1>
      </PostCardContainer>
    )
  }
}

export default PostCards
