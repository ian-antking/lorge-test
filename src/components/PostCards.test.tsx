import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import PostCards from './PostCards'

import '@testing-library/jest-dom'

interface Post {
  id: string
  title: string
  author: string
  blurb: string
  published: string
}

describe('PostCards', () => {
  it('displays the correct number of PostCard elements', () => {
    const data: Post[] = [
      {
        id: 'id1',
        title: 'title1',
        published: new Date(Date.now()).toISOString(),
        blurb: 'blurbText1',
        author: 'authorText1'
      },
      {
        id: 'id2',
        title: 'title2',
        published: new Date(Date.now()).toISOString(),
        blurb: 'blurbText2',
        author: 'authorText2'
      }
    ]

    const { asFragment } = render(<PostCards posts={data} />, {
      wrapper: MemoryRouter
    })

    expect(asFragment().children[0].childNodes.length).toBe(2)
  })

  it('renders a message if there are no posts', () => {
    const { getByText } = render(<PostCards posts={[]} />, {
      wrapper: MemoryRouter
    })

    expect(getByText('Nothing Here')).toBeInTheDocument()
  })

  it('renders consistently', () => {
    const data: Post[] = [
      {
        id: 'id1',
        title: 'title1',
        published: '2022-04-18T18:11:54+0000',
        blurb: 'blurbText1',
        author: 'authorText1'
      },
      {
        id: 'id2',
        title: 'title2',
        published: '2022-04-18T18:11:54+0000',
        blurb: 'blurbText2',
        author: 'authorText2'
      }
    ]

    const { asFragment } = render(<PostCards posts={data} />, {
      wrapper: MemoryRouter
    })

    expect(asFragment()).toMatchSnapshot()
  })
})
