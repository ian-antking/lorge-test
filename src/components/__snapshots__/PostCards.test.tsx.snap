// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCards renders consistently 1`] = `
<DocumentFragment>
  <div
    class="sc-gKXOVf deyyRI"
  >
    <a
      class="sc-bczRLJ eyjeUM"
      href="/post/id1"
    >
      <div
        class="sc-gsnTZi cznRGl"
      >
        <div
          class="sc-hKMtZM hNxTqh"
        >
          <span>
            18th April 2022
          </span>
        </div>
        <div
          class="sc-dkzDqf jGIDpV"
        >
          <span>
            title1
          </span>
        </div>
        <div
          class="sc-eCYdqJ eTHGuY"
        >
          blurbText1
        </div>
        <div
          class="sc-jSMfEi jNHCZO"
        >
          <span>
            authorText1
          </span>
        </div>
      </div>
    </a>
    <a
      class="sc-bczRLJ eyjeUM"
      href="/post/id2"
    >
      <div
        class="sc-gsnTZi cznRGl"
      >
        <div
          class="sc-hKMtZM hNxTqh"
        >
          <span>
            18th April 2022
          </span>
        </div>
        <div
          class="sc-dkzDqf jGIDpV"
        >
          <span>
            title2
          </span>
        </div>
        <div
          class="sc-eCYdqJ eTHGuY"
        >
          blurbText2
        </div>
        <div
          class="sc-jSMfEi jNHCZO"
        >
          <span>
            authorText2
          </span>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
